packages/vuetify/src/labs/VDateInput/VDateInput.tsx

```diff
@@ -232,8 +235,14 @@ export const VDateInput = genericComponent<VDateInputSlots>()({
                           } else {
                             model.value = value
 
-                            if (!props.multiple) {
-                              menu.value = false
+                            const isRange = props.multiple === 'range'
+                            const isSingle = props.multiple !== true && !isRange
+
+                            if (isSingle) {
+                              menu.value = props.closeOnSelection
```

Comment:
Looks like it should be:

```suggestion
                              menu.value = !props.closeOnSelection
```
?